### PR TITLE
Add annotation for endpoints to indicate minimum supported API version

### PR DESCRIPTION
**Pull request by t8y8**
Addresses #55 

This is just the decorator, I manually tested with the `users.get` endpoint and a script.

Once we add the annotation to the actual endpoints this will get automagically covered by all existing tests. I can make a unittest that mocks everything to test this method, but that doesn't feel super useful (It'd literally be testing does `<` work).

I think I'd like to break it into two PRs, add this method first, and in the second PR add the annotations and and ensure existing tests still pass

Open to feedback on the exception name and where I put this method -- but endpoint felt like the most natural place for it

/cc @RussTheAerialist @LGraber

### CODE REVIEW

File: tableauserverclient/server/endpoint/endpoint.py [hunk 1]
```diff
@@ -69,3 +70,33 @@ def post_request(self, url, xml_request, content_type='text/xml'):
                                   content=xml_request,
                                   auth_token=self.parent_srv.auth_token,
                                   content_type=content_type)
+
+
+def api(*, version):
+    '''Annotate the minimum supported version for an endpoint.
+
+    Checks the version on the server object and does a float comparison.
+    It will raise an exception if the server vesion is > the version specified.
```
Comment (t8y8) [inline tableauserverclient/server/endpoint/endpoint.py]: 'vesion' -> 'version' typo

File: tableauserverclient/server/endpoint/endpoint.py [hunk 2]
```diff
@@ -69,3 +70,33 @@ def post_request(self, url, xml_request, content_type='text/xml'):
...
+    Example:
+    >>> @api(version=2.3)
+    >>> def get(self, req_options=None):
+    >>>     ...
+    '''
+    def _decorator(func):
+        @wraps(func)
+        def wrapper(self, *args, **kwargs):
+            server_version = self.parent_srv.version
+            if float(server_version) < version:
```
Comment (RussTheAerialist) [inline tableauserverclient/server/endpoint/endpoint.py]: Please use https://docs.python.org/3.6/distutils/apiref.html#module-distutils.version to compare version numbers.  Casting to float will break if/when we decide to be more nuanced with your version numbers.

File: tableauserverclient/server/endpoint/users_endpoint.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from .endpoint import Endpoint
+from .endpoint import Endpoint, api
```
Comment (RussTheAerialist) [inline tableauserverclient/server/endpoint/users_endpoint.py]: Do you actually use api anywhere?

Comment (t8y8) [inline tableauserverclient/server/endpoint/users_endpoint.py]: oops, this was left over from my manual test script

**Pull request merged**